fn/map.go

```diff
@@ -0,0 +1,28 @@
+package fn
+
+import (
+	"fmt"
+
+	"golang.org/x/exp/maps"
+)
+
+// KeySet converts a map into a Set containing the keys of the map.
+func KeySet[K comparable, V any](m map[K]V) Set[K] {
+	return NewSet(maps.Keys(m)...)
+}
+
+// NewSubMap creates a sub-map from a given map using specified keys. It errors
+// if any of the keys is not found in the map.
+func NewSubMap[K comparable, V any](m map[K]V, keys []K) (map[K]V, error) {
+	result := make(map[K]V)
+	for _, k := range keys {
+		v, ok := m[k]
+		if !ok {
+			return nil, fmt.Errorf("NewSubMap: missing key %v", k)
```

Comment:
I could see use cases where this might not be an error and the key should just be ignored instead, if it doesn't appear in the map. So maybe we'd also add a version of this function that just ignores non-found keys? Could perhaps call it `NewSubMapIntersect` or whatever.